config/internal/apiserver/deployment.yaml.tmpl

```
@@ -18,6 +18,54 @@ spec:
         component: data-science-pipelines
     spec:
       containers:
+        {{ if .APIServer.EnableRoute }}
+        - name: oauth-proxy
+          args:
+            - --https-address=:8443
+            - --provider=openshift
+            - --openshift-service-account=ds-pipeline-{{.Name}}
+            - --upstream=http://localhost:8888
+            - --tls-cert=/etc/tls/private/tls.crt
+            - --tls-key=/etc/tls/private/tls.key
+            - --cookie-secret=SECRET
+            - '--openshift-delegate-urls={"/": {"group":"route.openshift.io","resource":"routes","verb":"get","name":"ds-pipeline-{{.Name}}","namespace":"{{.Namespace}}"}}'
+            - '--openshift-sar={"namespace":"{{.Namespace}}", "resource": "routes", "resourceName":"ds-pipeline-{{.Name}}", "verb": "get"}'
```

The openshift-sar is used for user authentication,
without the resource group specified, it would think of default resource group "", routes is no longer in default resource group, this would cause any non cluster admin, to have difficulty reaching the service.
```suggestion
            - '--openshift-sar={"namespace":"{{.Namespace}}", "resource": "routes", "resourceName":"ds-pipeline-{{.Name}}", "verb": "get", "resourceAPIGroup":"route.openshift.io"}'
```

or we remove completely the openshift-sar and rely on openshift-delegate, as API server would any way be reached via token most of the times.